lib/manageiq/loggers/container.rb

```diff
@@ -63,8 +63,12 @@ def translate_error(level)
           SEVERITY_MAP[level] || "unknown"
         end
 
+        def request_id_from_extra_params
+          Thread.current[:extra_params].try(:[], :request_id)
+        end
+
         def request_id
-          Thread.current[:current_request]&.request_id
+          Thread.current[:current_request]&.request_id || request_id_from_extra_params
```

Comment:
I'm confused. Why not use the existing implementation? Or if you don't want to do that, why add a separate key rather than making this line something like this?
```ruby
Thread.current[:current_request]&.request_id || Thread.current[:current_request].try(:[], :request_id)
```

Reply:
>> I'm confused. Why not use the existing implementation? 

I don't want to confuse programmers when we are populating `Thread.current[:current_request]`  with [other type](https://github.com/lpichler/manageiq-api-common/blob/0ae95dc1147d9a04e5f349f7bb4f6f6c8328f45c/lib/manageiq/api/common/request.rb#L32) then it would be used here(with hash) but probably I already confused you :D 

My goal is to pass `request_id` here in services where `request_id` is not present in [request](https://github.com/lpichler/manageiq-api-common/blob/0ae95dc1147d9a04e5f349f7bb4f6f6c8328f45c/lib/manageiq/api/common/request.rb#L32) but [parsed from Kafka](https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/75/files#diff-78c91e3cadb2cb4cb850c32197706daeR22).

But if you think that it is OK and I can use 
`Thread.current[:current_request]` as hash with key `request_id` in dependent PR.

Reply:
but OK I updated it as you suggested 👍

Reply:
Sorry, I didn't get a chance to respond to this ... 

I guess my real preference would be that we switch from assuming a ManageIQ::API Request object and enforce that everything is a hash. But this is a fine middle ground until we can change the miq api.

Reply:
I'd prefer to have one place to look for the request id.  What do you think of `Thread.current[:request_id]`?

Reply:
Oh, I like that idea, but I'm trying to balance how clean this gem's implementation is vs disruption to MIQ.

I guess my question is do we want to take on changing wherever this is used in the rest of manageiq now or later?

Reply:
We should probably support both and add a deprecation warning for the old location and release a minor version.  Then we can drop the old one and release a major version.

Reply:
done 👍 and I added deprecated warning.
thanks